src/main/java/org/opentripplanner/updater/trip/TimetableSnapshotSource.java

```diff
@@ -297,8 +310,18 @@ public UpdateResult applyTripUpdates(
                 tripId,
                 serviceDate
               );
-              case CANCELED -> handleCanceledTrip(tripId, serviceDate, CancelationType.CANCEL);
-              case DELETED -> handleCanceledTrip(tripId, serviceDate, CancelationType.DELETE);
+              case CANCELED -> handleCanceledTrip(
+                tripId,
+                serviceDate,
+                CancelationType.CANCEL,
+                canceledPreviouslyAddedTrip
+              );
+              case DELETED -> handleCanceledTrip(
+                tripId,
+                serviceDate,
+                CancelationType.DELETE,
+                canceledPreviouslyAddedTrip
+              );
```

Comment:
I'm not too happy about passing down a boolean to method which leads to an immediate return and I would like to handle it earlier.

In an ideal world Java would have a syntax like this:

```
case CANCELED if canceledPreviouslyAddedTrip -> Result.success(UpdateSuccess.noWarnings());
```
but it doesn't.

How about catching that case before the switch? So something like

```
if (CANCELLED or DELETED) and canceledPreviouslyAddedTrip return Result.success(UpdateSuccess.noWarnings());
```

Reply:
In generate I agree with you. However, the handleCanceledTrip is where you expect the handling of any valid case of a CANCELED scheduled relationship be.